sklearn/utils/_metadata_requests.py

```diff
@@ -0,0 +1,1157 @@
+        res = Bunch()
+        if self._self:
+            res.update(self._self._route_params(params=params, method=method))
+
+        param_names = self._get_param_names(
+            method=method, return_alias=True, ignore_self=True
+        )
+        child_params = {
+            key: value for key, value in params.items() if key in param_names
+        }
+        for key in set(res.keys()).intersection(child_params.keys()):
+            # conflicts are okay if the passed objects are the same, but it's
+            # an issue if they're different objects.
+            if child_params[key] is not res[key]:
+                raise ValueError(
+                    f"In {self.owner}, there is a conflict on {key} between what is"
+                    " requested for this estimator and what is requested by its"
+                    " children. You can resolve this conflict by using an alias for"
+                    " the child estimator(s) requested metadata."
+                )
+
+        res.update(child_params)
+        return res
+
+    def route_params(self, *, caller, params):
+        """Return the input parameters requested by child objects.
+
+        The output of this method is a bunch, which includes the inputs for all
+        methods of each child object that are used in the router's `method`.
+
+        If the router is also a consumer, it also checks for warnings of
+        `self`'s/consumer's requested metadata.
+
+        Parameters
+        ----------
+        caller : str
+            The name of the method for which the parameters are requested and
+            routed. If called inside the :term:`fit` method of a router, it
+            would be `"fit"`.
+
+        params : dict
+            A dictionary of provided metadata.
+
+        Returns
+        -------
+        params : Bunch
+            A :class:`~utils.Bunch` of the form
+            ``{"object_name": {"method_name": {prop: value}}}`` which can be
+            used to pass the required metadata to corresponding methods or
+            corresponding child objects.
+        """
+        if self._self:
+            self._self._check_warnings(params=params, method=caller)
+
+        res = Bunch()
+        for name, route_mapping in self._route_mappings.items():
+            router, mapping = route_mapping.router, route_mapping.mapping
+
+            res[name] = Bunch()
+            for _callee, _caller in mapping:
+                if _caller == caller:
+                    if router.type == "request":
+                        res[name][_callee] = router._route_params(
+                            params=params, method=_callee
+                        )
+                    else:
+                        res[name][_callee] = router._get_squashed_params(
+                            params=params, method=_callee
+                        )
+        return res
+
+    def validate_metadata(self, *, method, params):
+        """Validate given metadata for a method.
```

Comment:
Hopefully not, but I remember while working on https://github.com/scikit-learn/scikit-learn/pull/20350 there were such cases. I'm note too bothered about it since they're coupled together in `process_routing`. And yes, the cross validation example seems like a potential use-case.